streaming/streaming.go

```diff
@@ -459,9 +459,52 @@ func LaunchDataflowJob(ctx context.Context, targetProfile profiles.TargetProfile
 			dataflowSubnetwork = fmt.Sprintf("https://www.googleapis.com/compute/v1/projects/%s/regions/%s/subnetworks/%s", dataflowHostProjectId, dataflowCfg.Location, dataflowCfg.Subnetwork)
 		}
 	}
-
-	launchParameters := createLaunchParameters(dataflowCfg, inputFilePattern, project, datastreamCfg, instance, dbName, streamingCfg, dataflowSubnetwork, workerIpAddressConfig)
-
+	fmt.Printf("Working with dfCfg: %v\n", dataflowCfg)
+	maxWorkers := int32(50)
+	numWorkers := int32(1)
+	if dataflowCfg.MaxWorkers != "" {
+		intVal, err := strconv.ParseInt(dataflowCfg.MaxWorkers, 10, 64)
+		if err != nil {
+			return fmt.Errorf("could not parse MaxWorkers parameter %s, please provide a positive integer as input", dataflowCfg.MaxWorkers)
+		}
```

Comment:
From the UI/CLI flow, can't the validations be performed upfront instead of at the orchestration stage? 
Seems like sanitised values should be passed to this layer.

Reply:
We do perform some level of validations at the UI layer. But the backend should still be robust and have handling, especially given we have so many ways to reach here (UI sharded, non sharded, cli). Having prechecks help fail fast but think its best to keep our backend fullproof i.e. ensure job launch means params are right as much as possible.

Reply:
I don't think thats a good way to handle this, since it would mean every layer of the code needs to have these type of validations.
Validating input parameters at the top request layer and passing sanitised inputs to the underlying execution layers is a pretty standard practice. 

Since we are parsing integers here, this is the first instance of parsing validation logic we have had to add here. I suggest we think deeper about this.